Religious Doctrines

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Moved Mord'truari to Deathbringer religion and Sargerite to Doombringer religion.
- Implemented [Religious Hostility](https://docs.google.com/spreadsheets/d/1sv1AtanmOCDWhOYOrMU1NLu5F9WhWSYPIF37g7bB4YE/edit?usp=sharing).
- Assigned doctrines and tenets to all the religions.
- Assigned sins and virtues to all the religions.
- **New doctrines:**
  - Added All Means to End doctrine that allows aggressive conquests, sacrifices and tyranny. Used by Shath'gral (Old Gods' religion), Doombringer (Scourge's religion), Deathbringer (Legion's religion) religions.
- **New tenets:**
  - Added 6 Syncheritc tenets. One per a religious family.
  - Added Cult of Strenght tenet. Used by Fetishistic, Roanaki, Y'Shaarji, Krosi faiths.
  - Added Ritual Hunt tenet. Used by Am'atali faith.
  - Added Progressivism tenet. Used by Pantheonic, Shenic, Suntouched, Rationalistic faiths.
  - Added Consumerism tenet. Used by Fortunistic faith.
  - Added Moon Worship tenet. Used by Elunic faith.

## How to test:
Check religions, tell your opinion about their tenets and doctrines. If you have tenet ideas, feel free to suggest. I may do them now if they're easy or write them down to implement them later.

BronzeBaron commented:
Noticed that Odynism has human sacrifice, maybe cult of strength instead?

Looked through all the religions, many of them seem to have tenets/doctrines that are quite fitting. 

Doctrine idea; if culling provinces for culture/faith is still a thing, perhaps a doctrine that makes it reward piety or prestige and encourages the AI to do it more often?

zumbak04 commented:
@BronzeBaron 
> Noticed that Odynism has human sacrifice, maybe cult of strength instead?

Done in 5cb1478e35124bf5f042dcd0889465692253f7d8.
> Doctrine idea; if culling provinces for culture/faith is still a thing, perhaps a doctrine that makes it reward piety or prestige and encourages the AI to do it more often?

No, but there is culture and religion converting. Who will have it?

arithon reviewed (commented):
loc LGTM so far, just some small ideas

MemeWizard commented on localization/english/religion/wc_core_tenets_l_english.yml:
```diff
@@ -0,0 +1,69 @@
+ tenet_dharmic_pacifism_name_neutral:0 "Chi Pacifism"
+ 
+ special_doctrine_all_means_to_end_name:0 "All Means to End"
+ special_doctrine_all_means_to_end_desc:0 "While others think what is right and what is wrong, we act. Our goal is righteous. Our will is unshakable. From the ashes shall arise a new order that will shake the very foundations of the world."
```
It's two different things.

MemeWizard commented:
Can I recommend adding comments near changes in vanilla files?

zumbak04 commented on localization/english/religion/wc_core_tenets_l_english.yml:
```diff
@@ -0,0 +1,69 @@
+﻿l_english:
+ tenet_death_syncretism_name:0 "Death Syncretism"
+ tenet_death_syncretism_desc:0 "There is something to be learned from Death cults. They may not follow the true faith, but that does not make them evil."
```
Done in 413ab6490a8fd29ce84799fd3b85ace65c8d9d6c.

zumbak04 commented on localization/english/religion/wc_core_tenets_l_english.yml:
```diff
@@ -0,0 +1,69 @@
+ doctrine_parameter_hostility_override_disorder_hostility_doctrine:0 "Considers Disorder [faiths|E], and is considered by Disorder [faiths|E], to be [GetHostilityLevelName( FAITH_DOCTRINE_PARAMETER.GetIntValue )|V]"
+ 
+ tenet_elemental_syncretism_name:0 "Elemental Syncretism"
+ tenet_elemental_syncretism_desc:0 "There is something to be learned from Elemental cults. They may not follow the true faith, but that does not make them evil."
```
Done in 413ab6490a8fd29ce84799fd3b85ace65c8d9d6c.

zumbak04 commented on localization/english/religion/wc_core_tenets_l_english.yml:
```diff
@@ -0,0 +1,69 @@
+ doctrine_parameter_hostility_override_death_hostility_doctrine:0 "Considers Death [faiths|E], and is considered by Death [faiths|E], to be [GetHostilityLevelName( FAITH_DOCTRINE_PARAMETER.GetIntValue )|V]"
+ 
+ tenet_disorder_syncretism_name:0 "Disorder Syncretism"
+ tenet_disorder_syncretism_desc:0 "There is something to be learned from Disorder or Fel cults. They may not follow the true faith, but that does not make them evil."
```
Done in 413ab6490a8fd29ce84799fd3b85ace65c8d9d6c.

zumbak04 commented:
> Can I recommend adding comments near changes in vanilla files?

I believe I added everywhere except find&replace changes.

MemeWizard commented:
`Cult of Strength` do not have `strong` trait as a virtue.

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -350,72 +384,79 @@
+						# has_doctrine = pagan_hostility_doctrine
+					# }
+				# }
+				# value = faith_tenet_cost_high
+			# }
+			# else = {
+				# value = faith_tenet_cost_mid
+			# }
+
+			# # Multiplier for keeping same tenet
+			# if = {
+				# limit = { has_doctrine = tenet_gnosticism }
+				# multiply = faith_unchanged_doctrine_cost_mult
+			# }
+		# }
+
+		# traits = {
+			# virtues = { temperate }
+			# sins = { gluttonous }
+		# }
+
+		# can_pick = {
+			# custom_description = {
+				# text = incompatible_tenet_eastern_syncretism_trigger
+				# NOT = {
+					# flag:tenet_eastern_syncretism = { is_in_list = selected_doctrines }
+				# }
+			# }
+		# }
+
+		# parameters = {
+			# hostility_override_tenet_gnosticism = 0
+			# granting_titles_gives_stress = yes
+		# }
+
+		# character_modifier = {
+			# learning = 2
+			# stewardship = -2
+		# }
+	# }
```
I suggest keeping it simple and just delete it if we are not going to use it, leaving a comment about what it was
```suggestion
	# Warcraft
	# tenet_gnosticism
```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -350,72 +384,79 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -467,7 +508,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -516,7 +557,7 @@
 			if = {
 				limit = {
-					religion_tag = christianity_religion
+					like_christianity_religion_trigger = yes
```
```suggestion

					# Warcraft
					like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -582,7 +621,7 @@
 				triggered_desc = {
 					trigger = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -594,7 +633,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -670,11 +709,17 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -767,7 +815,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -823,7 +871,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -985,7 +1034,7 @@
 		is_shown = {
 			NOR = {
-				religion_tag = christianity_religion
+				like_christianity_religion_trigger = yes
```
```suggestion

				# Warcraft
				like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1003,7 +1052,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft				
						like_christianity_religion_trigger = yes
```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1907,127 +2002,205 @@
 	#############################
 	# Syncretic Faith Tenets	#
 	#############################
```
I suggest moving them  to a separate file

zumbak04 replied:
See https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/21/files#r518731788.

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -2529,5 +2820,207 @@
 	#	}
 	#}
+	
+	##############################
+	#	Warcraft
+	##############################
```
I suggest creating our own file with tenets instead

zumbak04 replied:
I tried, it didn't work. The game has seen them as separate tenets.

MemeWizard commented on common/religion/religions/lizard_group.txt:
```diff
@@ -1,39 +1,47 @@
 	traits = {
-		virtues = { forgiving compassionate brave }
-		sins = { vengeful sadistic craven }
+		virtues = {
+			ambitious
+			brave
+			diligent
+		}
+		sins = {
+			content
+			craven
+			lazy
+		}
```
Just to keep it consistent
```suggestion
		virtues = { ambitious brave diligent }
		sins = { content craven lazy }
```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1136,7 +1185,7 @@
 
 		is_shown = {
-			NOT = { religion_tag = christianity_religion }
+			NOT = { like_christianity_religion_trigger = yes }
```
```suggestion

			# Warcraft
			NOT = { like_christianity_religion_trigger = yes }

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1150,7 +1199,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1214,7 +1269,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1268,9 +1321,18 @@
 
 	tenet_dharmic_pacifism = { #Mix of reincarnation and pacifism rites
 		icon = core_tenet_dharmawheel
+		
+		# Warcraft
+		name = {
+			first_valid = {
+				desc = tenet_dharmic_pacifism_name_neutral
+			}
+		}
+		
```
To replace vanilla localization you need to place the .yml file in `/localization/replace/<language>/`

zumbak04 replied:
Done in 9fb485782b7c952468bd6db35a94a8aded652f76.

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1328,7 +1390,7 @@
 
 		is_shown = {
-			has_doctrine = eastern_hostility_doctrine
+			like_eastern_religion_trigger = yes
```
```suggestion

			# Warcraft
			like_eastern_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1478,7 +1542,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1520,7 +1589,7 @@
 				limit = {
 					OR = {
-						religion_tag = christianity_religion
+						like_christianity_religion_trigger = yes
```
```suggestion

						# Warcraft
						like_christianity_religion_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1605,14 +1674,15 @@
 			# Multiplier for keeping same tenet
 			if = {
-				limit = { has_doctrine = tenet_human_sacrifice }
+				limit = { has_doctrine_tenet_human_sacrifice_trigger = yes }
```
```suggestion

				# Warcraft
				limit = { has_doctrine_tenet_human_sacrifice_trigger = yes }

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1605,14 +1674,15 @@
 		}
 
 		is_shown = {
+			evil_public_religion_trigger = no	#They have it by default
```
```suggestion

			# Warcraft
			evil_public_religion_trigger = no	#They have it by default
```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1816,11 +1903,13 @@
 		}
 	}
 
+	# Warcraft
+	#Includes tenet_light_syncretism
 	tenet_sun_worship = {
```
```suggestion
	tenet_sun_worship = { # Warcraft - Includes tenet_light_syncretism
```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1830,20 +1919,23 @@
 
 		is_shown = {
-			religion_tag = zunism_religion
+			evil_public_religion_trigger = no
+			
+			tenet_sun_worship_allowed_trigger = yes
```
```suggestion

			# Warcraft
			evil_public_religion_trigger = no
			
			tenet_sun_worship_allowed_trigger = yes

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1830,20 +1919,23 @@
 			summer_festivals_active = yes
 			trial_by_sun_active = yes
+			
+			#tenet_light_syncretism
+			hostility_override_light_hostility_doctrine = 1
```
```suggestion
			# Warcraft - tenet_light_syncretism
			hostility_override_light_hostility_doctrine = 1

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1865,14 +1957,17 @@
 			# Multiplier for keeping same tenet
 			if = {
-				limit = { has_doctrine = tenet_warmonger }
+				limit = { has_doctrine_tenet_warmonger_trigger = yes }
```
```suggestion

				# Warcraft
				limit = { has_doctrine_tenet_warmonger_trigger = yes }

```

MemeWizard commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -1865,14 +1957,17 @@
+
 			NOR = {
-				religion_tag = christianity_religion
+				like_christianity_religion_trigger = yes
```
```suggestion

				# Warcraft
				like_christianity_religion_trigger = yes

```

MemeWizard reviewed (changes requested):
code comments pls

Also:

<details>
<summary>Can't reform Murloc religion because ritual cannibalism is not allowed</summary>

![image](https://user-images.githubusercontent.com/46576860/98304369-e7f23180-1fd0-11eb-83fc-1237631598ae.png)

</details>

MemeWizard commented:
> All Means to End 

Shouldn't it be "All Means to **an** End"?

zumbak04 commented on common/decisions/30_activity_decisions.txt:
```diff
@@ -194,6 +200,26 @@ start_hunt_decision = {
+
+				min = medium_piety_value # Always give some piety, even if we don't have any of the specified tenets.
+			}
+		}
```
Done in 2e9d9194b314dc3fcdbf22cad84865f57282c00f.

zumbak04 commented on common/scripted_triggers/00_war_and_peace_triggers.txt:
```diff
@@ -560,7 +560,7 @@ desirable_for_capture_trigger = {
 can_use_viking_invasion_cbs_trigger = {
 	faith = {
-		has_doctrine = tenet_warmonger
+		has_doctrine_tenet_warmonger_trigger = yes
```
(GDC)

zumbak04 commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -8,11 +8,24 @@
+				desc = tenet_aniconism_desc_neutral
+			}
+		}
+		
```
Done in 9fb485782b7c952468bd6db35a94a8aded652f76.

zumbak04 commented on common/religion/doctrines/00_core_tenets.txt:
```diff
@@ -48,6 +62,19 @@
+				desc = tenet_alexandrian_catechism_desc_neutral
+			}
+		}
+		
```
Done in 9fb485782b7c952468bd6db35a94a8aded652f76.

zumbak04 commented:
> `Cult of Strength` do not have `strong` trait as a virtue.

Fixed in a6bf286c21848e9bc51b3af41cf74d86d4b05b21.

> Can't reform Murloc religion because ritual cannibalism is not allowed

Fixed in 72d675edacae15680fc294c5ab1ead984933714a.

> Shouldn't it be "All Means to **an** End"?

We avoid articles in names.

TheGroxEmpire commented:
- Ancestor Worship is not allowed for reform tenets:

<details>

![image](https://user-images.githubusercontent.com/47816228/98622769-db593a80-233c-11eb-87b5-37dcbe73216b.png)
</details>

- Maybe replace Communal Identity tenet in Rationalism with Critical Approach
- Suggestion: Replace Eastern Syncretism with Religious Tolerance which consider most faith Righteous / Astray except for truly destructive religion like Shath'gral, Doombringer, Fel, etc. I think it'd be nice to have for big faction like Alliance or Horde down the road. Suggested flavor text:
> "We must be tolerant yet rigid in our beliefs. Anything taken to an extreme is hazardous to our world." (https://classic.wowhead.com/quest=5251/the-archivist)

MemeWizard commented:
> We avoid articles in names.

Even if they are used in vanilla?

<details>
<summary>The</summary>

![image](https://user-images.githubusercontent.com/46576860/98627077-6548d980-2324-11eb-945d-1d2545b78d97.png)

</details>

TheGroxEmpire reviewed (commented):
- Ancestor Worship is not allowed for reform tenets:

<details>

![image](https://user-images.githubusercontent.com/47816228/98622769-db593a80-233c-11eb-87b5-37dcbe73216b.png)
</details>

- Maybe replace Communal Identity tenet in Rationalism with Critical Approach
- Tenet suggestion: Replace Eastern Syncretism with Religious Tolerance which consider most faith Righteous / Astray except for truly destructive religion like Shath'gral, Doombringer, Fel, etc. I think it'd be nice to have for big faction like Alliance or Horde down the road. Suggested flavor text:
> "We must be tolerant yet rigid in our beliefs. Anything taken to an extreme is hazardous to our world." (https://classic.wowhead.com/quest=5251/the-archivist)

zumbak04 commented:
> Even if they are used in vanilla?

"The" is used sometimes. Especially if it's a proper name like the Roman Empire.
Paradox doesn't follow any rules, but most of the decisions skip articles.
Our rules are [Here](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/Guidelines-(For-CK3)#localization-guidelines).

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/98822527-64aa6300-244a-11eb-8a24-d5d93a5086eb.png)

</details>

zumbak04 commented:
> 
> * Ancestor Worship is not allowed for reform tenets:

Fixed in eb4ab6f287e7c9d802ecc5991985f18225886b9f.

> * Maybe replace Communal Identity tenet in Rationalism with Critical Approach

Done in 9e837a2a33d70a0fe4ecf73b818881d977b2feb0.

> * Tenet suggestion: Replace Eastern Syncretism with Religious Tolerance which consider most faith Righteous / Astray except for truly destructive religion like Shath'gral, Doombringer, Fel, etc. I think it'd be nice to have for big faction like Alliance or Horde down the road. Suggested flavor text:

This will block holy wars between these religions. So, if the elves consider Loa faiths Astray, they won't be able to attack the Amani with holy wars.

TheGroxEmpire reviewed (approved):
> This will block holy wars between these religions. So, if the elves consider Loa faiths Astray, they won't be able to attack the Amani with holy wars.

I guess that's an issue yeah. I'll get back to it if I find a better way to do this.

MemeWizard commented:
Add missing localization

<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/98914127-548d9480-24d9-11eb-9842-65c1df612cdd.png)

</details>

zumbak04 commented:
> Add missing localization

Fixed in a59d6461fdad66caa41e4a702be2d158a59c8b8d.

MemeWizard commented:
1

> Holy light - clerical gender - only men

It should be "either", innit?

https://wow.gamepedia.com/List_of_priests

2
Wrong icon:

> [religion_templates.cpp:167]: 'gorgog_worship' is an unreformed faith so must have a reformed icon set

3
> Paradox doesn't follow any rules, 

I know that. But we do not have any reason to be as bad as Paradox, do we?

> but most of the decisions skip articles.

Ok, but I was talking about doctrines, not about decisions,

> Our rules are Here.

Rules that prohibit using correct language does not make it even slightly better.

Moreover, article here is important. "All means to end" and "All means to **an** end" is not the same thing.
The first one is an incomplete sentence. All means to end *what*? All means to end the first war? All means to end the scourge invasion?
Second has more clear meaning.

If you prefer "**the**" more than "**an**", then I suggest using `The outcome justifies the deeds` or `the end justifies the means` instead

MemeWizard commented on common/customizable_localization/00_ghw_tenet_custom_loc.txt:
```diff
@@ -19,8 +19,11 @@ ghw_from_which_tenet = {
+			faith = { has_doctrine_tenet_warmonger_trigger = yes }
+
 		}
 		localization_key = ghw_from_which_tenet_warmonger
```
> # GHWs given by which tenet?

> localization_key = ghw_from_which_tenet_warmonger

>  ghw_from_which_tenet_warmonger:0 "$tenet_warmonger_name$"

It will be shown that GHW is given by warmonger tenet, despite the fact that it is given by special doctrine,

zumbak04 replied:
Done in 630576164d7f975ec1719315a94bacb554618712.

zumbak04 commented:
> It should be "either", innit?

Done in 4d44430dc50b644a7fd7d8dd3a8f4c5836ca10ec.

> Wrong icon:

Fixed in 971dae6e94dde51a26643edf45da78a5220f1ffb.

> If you prefer "**the**" more than "**an**", then I suggest using `The outcome justifies the deeds` or `the end justifies the means` instead

I like. I skipped the because it's not important to the meaning in e26b75fd6d529a25b85d02f079dd41d0fd27b6e8.

zumbak04 commented on common/religion/doctrines/wc_doctrines_special.txt:
```diff
@@ -0,0 +1,166 @@
+﻿#So far not used because we use hostility doctrines instead
```
Done in 4e0fb674f292b5d053c2fb99fafb1d7ed1398252.

MemeWizard reviewed (approved):
ok, but

zumbak04 commented on common/scripted_triggers/wc_religious_triggers.txt:
```diff
@@ -53,4 +53,116 @@ like_baltic_religion_trigger = {
+#Kinda allowed trigger
+faith_likes_hunt_trigger = {
+	OR = {
+		religion_tag = loa_group
+		religion = { is_in_family = rf_elemental }
```
Done in dc2bca90f4125c9cdac699452ac936f3f0d07cd8.
